node/node.go

```diff
@@ -107,6 +108,12 @@ func NewNode(config cfg.Config, privValidator *types.PrivValidator, clientCreato
 	}
 	consensusReactor := consensus.NewConsensusReactor(consensusState, fastSync)
 
+	// Transaction indexing
+	store := dbm.NewDB("tx_indexer", config.GetString("db_backend"), config.GetString("db_dir"))
+	indexer := txindexer.NewKV(store)
+	bcReactor.SetTxIndexer(indexer)
```

Comment:
I am curious what happens when we copy something holding a reference to levelDB (with open file descriptor). Do copy shares the same file descriptor with original object?

```
func (s *State) Copy() *State {
	return &State{
		db:              s.db,
		GenesisDoc:      s.GenesisDoc,
		ChainID:         s.ChainID,
		LastBlockHeight: s.LastBlockHeight,
		LastBlockID:     s.LastBlockID,
		LastBlockTime:   s.LastBlockTime,
		Validators:      s.Validators.Copy(),
		LastValidators:  s.LastValidators.Copy(),
		AppHash:         s.AppHash,
		TxIndexer:       s.TxIndexer,
	}
}
```